### PR TITLE
GCS deleteBlobsIgnoringIfNotExists should catch StorageException

### PR DESCRIPTION
**Pull request by tlrx**
`GoogleCloudStorageBlobStore.deleteBlobsIgnoringIfNotExists()` does not correctly catch `StorageException` thrown by `batch.submit()`.

In the case a snapshot is deleted through `BlobStoreRepository.deleteSnapshot()` a storage exception [is not caught](https://github.com/elastic/elasticsearch/compare/master...tlrx:gcs-catch-storage-exception?expand=1#diff-83e9ae765eb2a80bbbedd251b686cc10R410) (only `IOException` are) so the deletion is interrupted and indices cannot be cleaned up. The storage exception bubbles up to `SnapshotService.deleteSnapshotFromRepository()` but the [listener that removes the deletion from the cluster state is not executed](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java#L1347), leaving the deletion in the cluster state.

This bug has been reported in #46772 where `batch.submit()` threw an exception in the test `testIndicesDeletedFromRepository` and following tests failed because a snapshot deletion was running.

Relates #46772

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -407,7 +407,7 @@ public void deleteSnapshot(SnapshotId snapshotId, long repositoryStateId, Action
                 Arrays.asList(snapshotFormat.blobName(snapshotId.getUUID()), globalMetaDataFormat.blobName(snapshotId.getUUID()));
             try {
                 blobContainer().deleteBlobsIgnoringIfNotExists(snapMetaFilesToDelete);
-            } catch (IOException e) {
+            } catch (Exception e) {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Didn't you just make this change unnecessary by handling all exceptions and re-throwing them as `IOException`?

**Review by original-brownbear — APPROVED** [review]
LGTM thanks @tlrx ! just one question (but I'm fine with keeping the change commented on for safety's sake).

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java (hunk 1 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Arghl, this is a leftover, I'll revert this

**Comment by tlrx** [comment]
Thanks @original-brownbear

**Pull request merged**